eva-server/src/main/java/uk/ac/ebi/variation/eva/server/ws/ArchiveWSServer.java

```diff
@@ -63,54 +64,40 @@
     private StudyDBAdaptor studyDgvaDbAdaptor;
     private StudyDBAdaptor studyEvaproDbAdaptor;
     
-    public ArchiveWSServer() {
-        super();
-    }
-
-    public ArchiveWSServer(@Context UriInfo uriInfo, @Context HttpServletRequest hsr) 
-            throws NamingException, IOException {
-        super(uriInfo, hsr);
+    public ArchiveWSServer() throws NamingException, IOException {
         archiveDgvaDbAdaptor = new ArchiveDgvaDBAdaptor();
         archiveEvaproDbAdaptor = new ArchiveEvaproDBAdaptor();
         studyDgvaDbAdaptor = new StudyDgvaDBAdaptor();
         studyEvaproDbAdaptor = new StudyEvaproDBAdaptor();
     }
 
-    @GET
-    @Path("/files/count")
-    public Response countFiles() {
-        return createOkResponse(archiveEvaproDbAdaptor.countFiles());
+    @RequestMapping(value = "/files/count", method = RequestMethod.GET)
+    public QueryResponse countFiles() {
+        return setQueryResponse(archiveEvaproDbAdaptor.countFiles());
     }
-    
-    @GET
-    @Path("/species/count")
-    public Response countSpecies() {
-        return createOkResponse(archiveEvaproDbAdaptor.countSpecies());
+
+    @RequestMapping(value = "/species/count", method = RequestMethod.GET)
+    public QueryResponse countSpecies() {
+        return setQueryResponse(archiveEvaproDbAdaptor.countSpecies());
     }
-    @GET
-    @Path("/species/list")
-    public Response getSpecies() {
-        try {
-            Properties properties = new Properties();
-            properties.load(DBAdaptorConnector.class.getResourceAsStream("/eva.properties"));
-            
-            return createOkResponse(archiveEvaproDbAdaptor.getSpecies(properties.getProperty("eva.version"), true));
-        } catch (IOException ex) {
-            return createErrorResponse(ex);
```

Comment:
In case of exceptions thrown inside any method, Spring generates an error 500 with the exception message displayed (which are usually quite readable).